tests/dist/test_multi_partitioning.py

```diff
@@ -0,0 +1,77 @@
+import dgl
+
+import multiprocessing as mp
+import subprocess
+import os
+import unittest
+
+import pytest
+
+import ogb.lsc
+try:
+    from chunk_graph import chunk_graph
+except ImportError:
+    print("The test needs DIST_DGL_TEST_TOOLS_PATH in PYTHONPATH")
+    exit()
+
+
+def run(cmd):
+    subprocess.check_call(cmd, shell=True)
+
+tools_dir =  os.environ.get("DIST_DGL_TEST_TOOLS_PATH", "tools")
+ip_config = os.environ.get("DIST_DGL_TEST_IP_CONFIG", "ip_config.txt")
+ssh_port = os.getenv("DIST_DGL_TEST_SSH_PORT", "22")
+shared_workspace = os.environ.get('DIST_DGL_TEST_WORKSPACE', '/shared_workspace/mag240m/')
+local_workspace = os.environ.get('DIST_DGL_TEST_LOCAL_WORKSPACE', '/workspace')
+
+# When running the partitioning inside containers, Gloo may choose the wrong net interface
+# if the tests fails, set GLOO_SOCKET_IFNAME with the right interface in each container
+@unittest.skipIf(os.name == "nt", reason="Do not support windows yet")
+def test_multi_node_partitioning():
+    num_part =  sum(1 for _ in open(ip_config))
+
+    # Obtaining the dataset
+    dataset = ogb.lsc.MAG240MDataset(shared_workspace)
```

Comment:
could we use a small dataset to reproduce the bug? large dataset takes too much time for unit test.